src/api/Base.js

```diff
@@ -123,6 +124,75 @@ class Base {
     getCache(): Cache {
         return this.cache;
     }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} successCallback the success callback
+     */
+    successHandler(data: Object, successCallback: Function) {
```

Comment:
Few file already set successCallback/errorCallback on the `this` object, so I wonder maybe one can make this more generic, with the args just callback. Because I am assuming this is just to simplify the destroyed check.

```
successHandler(callback: Function) {
   if (!this.isDestroyed()) {
      callback();
    }
}
```


// Item.js
rename(...successCallback, errorCallback,...) {
        ...
        this.successCallback = successCallback;
        this.errorCallback = errorCallback;

        return this.xhr
            .put({ url: `${this.getUrl(id)}`, data: { name } })
            .then(this.successHandler(this.renameSuccessHandler))
            .catch(this.errorHandler);
}